pkg/controller/elasticsearch/client/official_log.go

```diff
@@ -0,0 +1,57 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package client
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/elastic/go-elasticsearch/v8/estransport"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var log = logf.Log.WithName("client")
+
+// esLogger implements estransport.Logger using the controller-runtime logging infrastructure
+type esLogger struct{}
+
+var (
+	_ estransport.Logger = &esLogger{}
+)
+
+// LogRoundTrip should not modify the request or response, except for consuming and closing the body.
+// Implementations have to check for nil values in request and response.
+func (l *esLogger) LogRoundTrip(
```

Comment:
Maybe you can (re)use the [`estransport.JSONLogger`](https://github.com/elastic/go-elasticsearch/blob/master/estransport/logger.go) here?

Reply:
The underlying logger does the serializing of the arguments JSON in our case, so I don't think the JSONLogger would be applicable (e.g resulting in double-serialized since we can't pass it already-seriailized data).

Reply:
The underlying logger we use would serialize the arguments to JSON, resulting in double-encoding instead of embedding some partial JSON. I suppose we could work around it by wrapping the JSONLogger result in a struct we provide a custom marshaller for, but it seems easier to do it this way for now

Reply:
Argh, right, I've missed that you pass a structure to the underlying logger. In this case a custom Logger implementation is definitely the right solution.